java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/media/ExifInterface;

<!-- Add [x] inside the brackets -->

- [x ] *I am using the latest version*
- [x ] *I can reproduce the bug in the demo app, and if not...*
- [x ] *I have tried to reproduce the bug in the demo app*

#### Detailed steps to reproduce the issue:

1.Target platform Lolipop 5.0
2.SDK API level 26
3.capturePicture()
#### Expected behavior:
Take a picture
#### Actual behavior:
Crashed
#### Screenshots:

#### Interesting logs:
`04-01 07:36:23.805 7945-7989/com.example.cqedu.lpr E/AndroidRuntime: FATAL EXCEPTION: FallbackCameraThread
                                                                     Process: com.example.cqedu.lpr, PID: 7945
                                                                     java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/media/ExifInterface;
                                                                         at com.otaliastudios.cameraview.CameraUtils.decodeBitmap(CameraUtils.java:114)
                                                                         at com.otaliastudios.cameraview.CameraUtils$1.run(CameraUtils.java:91)
                                                                         at android.os.Handler.handleCallback(Handler.java:819)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:164)
                                                                         at android.os.HandlerThread.run(HandlerThread.java:65)
                                                                      Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.media.ExifInterface" on path: DexPathList[[zip file "/data/app/com.example.cqedu.lpr-dewriNcKuVVlUrYGUKsoGg==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.cqedu.lpr-dewriNcKuVVlUrYGUKsoGg==/lib/x86, /system/lib]]
                                                                         at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
                                                                         at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                                         at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                         at com.otaliastudios.cameraview.CameraUtils.decodeBitmap(CameraUtils.java:114) 
                                                                         at com.otaliastudios.cameraview.CameraUtils$1.run(CameraUtils.java:91) 
                                                                         at android.os.Handler.handleCallback(Handler.java:819) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:164) 
                                                                         at android.os.HandlerThread.run(HandlerThread.java:65) `
> 
> `import android.support.media.ExifInterface;`
> This line in CameraUtils.java:11 seems to be wrong.It cannot find the "media" class. 
> `import android.media.ExifInterface;`
> This statement seems to be right.
> So, is it a SDK version problem?And which version should i use?
*You can use `CameraLogger.setLogLevel(LEVEL_VERBOSE)`
to add logs to logcat, and `CameraLogger.registerLogger()`
to handle logs yourself (e.g. export to file or a crash reporting service)
so you can collect data from your users.*

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+### v1.5.1
+
+- Bug: byte array length for Frames was incorrect thanks to [@ssakhavi][ssakhavi] ([#205][205])
+- Bug: gestures were crashing in some conditions ([#222][222])
+- Bug: import correctly the ExifInterface library ([#222][222])
+- Updated dependencies thanks to [@caleb-allen][caleb-allen] ([#190][190])
+
 ## v1.5.0
 
 - New: set encoder for video recordings with `cameraVideoCodec` ([#174][174])
@@ -80,6 +87,8 @@ https://github.com/natario1/CameraView/compare/v1.2.3...v1.3.0
 [RobertoMorelos]: https://github.com/RobertoMorelos
 [RocketRider]: https://github.com/RocketRider
 [xp-vit]: https://github.com/xp-vit
+[caleb-allen]: https://github.com/caleb-allen
+[ssakhavi]: https://github.com/ssakhavi
 
 [73]: https://github.com/natario1/CameraView/pull/73
 [80]: https://github.com/natario1/CameraView/pull/80
@@ -104,3 +113,6 @@ https://github.com/natario1/CameraView/compare/v1.2.3...v1.3.0
 [172]: https://github.com/natario1/CameraView/pull/172
 [173]: https://github.com/natario1/CameraView/pull/173
 [174]: https://github.com/natario1/CameraView/pull/174
+[190]: https://github.com/natario1/CameraView/pull/190
+[205]: https://github.com/natario1/CameraView/pull/205
+[222]: https://github.com/natario1/CameraView/pull/222

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ addressing most of the common issues and needs, and still leaving you with flexi
 See [CHANGELOG](https://github.com/natario1/CameraView/blob/master/CHANGELOG.md).
 
 ```groovy
-compile 'com.otaliastudios:cameraview:1.5.0'
+compile 'com.otaliastudios:cameraview:1.5.1'
 ```
 
 Make sure your project repositories include the `jcenter()`:

--- a/build.gradle
+++ b/build.gradle
@@ -7,7 +7,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.2.0-alpha15'
         // https://inthecheesefactory.com/blog/how-to-upload-library-to-jcenter-maven-central-as-dependency/en
         // https://www.theguardian.com/technology/developer-blog/2016/dec/06/how-to-publish-an-android-library-a-mysterious-conversation
         classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
@@ -25,7 +25,7 @@ allprojects {
 ext {
     compileSdkVersion = 27
     buildToolsVersion = "27.0.3"
-    supportLibVersion = '27.1.0'
+    supportLibVersion = '27.1.1'
     minSdkVersion = 15
     targetSdkVersion = 27
 }

--- a/cameraview/build.gradle
+++ b/cameraview/build.gradle
@@ -3,7 +3,7 @@ apply plugin: 'com.github.dcendents.android-maven'
 apply plugin: 'com.jfrog.bintray'
 
 // Required by bintray
-version = '1.5.0'
+version = '1.5.1'
 group = 'com.otaliastudios'
 
 //region android dependencies
@@ -48,7 +48,7 @@ dependencies {
     androidTestImplementation 'com.google.dexmaker:dexmaker-mockito:1.2'
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
 
-    implementation "com.android.support:exifinterface:$supportLibVersion"
+    api "com.android.support:exifinterface:$supportLibVersion"
     implementation "com.android.support:support-annotations:$supportLibVersion"
 }
 

--- a/cameraview/build.gradle
+++ b/cameraview/build.gradle
@@ -178,7 +178,7 @@ apply plugin: 'jacoco'
 
 def reportsDirectory = "$buildDir/reports/"
 jacoco {
-    toolVersion = "0.7.4+"
+    toolVersion = "0.8.1"
     reportsDir = file(reportsDirectory)
 }
 

--- a/cameraview/src/main/views/com/otaliastudios/cameraview/ScrollGestureLayout.java
+++ b/cameraview/src/main/views/com/otaliastudios/cameraview/ScrollGestureLayout.java
@@ -29,6 +29,7 @@ class ScrollGestureLayout extends GestureLayout {
             public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
                 boolean horizontal;
                 LOG.i("onScroll:", "distanceX="+distanceX, "distanceY="+distanceY);
+                if (e1 == null || e2 == null) return false; // Got some crashes about this.
                 if (e1.getX() != mPoints[0].x || e1.getY() != mPoints[0].y) {
                     // First step. We choose now if it's a vertical or horizontal scroll, and
                     // stick to it for the whole gesture.

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip

